aws/resource_aws_fsx_file_system_test.go

```diff
@@ -0,0 +1,365 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/fsx"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSFsxFileSystem_lustreBasic(t *testing.T) {
+	var v fsx.FileSystem
+	resourceName := "aws_fsx_file_system.test"
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
These acceptance tests are missing a [`CheckDestroy` function](https://www.terraform.io/docs/extend/testing/acceptance-tests/testcase.html#checkdestroy), which is used to verify from the API that the `Delete` function did actually delete the infrastructure. It can be implemented with something like the following:

```go
func testAccCheckFsxFileSystemDestroy(s *terraform.State) error {
	conn := testAccProvider.Meta().(*AWSClient).fsxconn

	for _, rs := range s.RootModule().Resources {
		if rs.Type != "aws_fsx_file_system" {
			continue
		}

		filesystem, err := describeFsxFileSystem(conn, rs.Primary.ID)

		if isAWSErr(err, fsx.ErrCodeFileSystemNotFound, "") {
			continue
		}

		if err != nil {
			return err
		}

		if filesystem != nil {
			return fmt.Errorf("FSx File System (%s) still exists", rs.Primary.ID)
		}
	}
	return nil
}
```

And in the test cases:

```go
		Providers:     testAccProviders,
		CheckDestroy:  testAccCheckFsxFileSystemDestroy,
		Steps: []resource.TestStep{
```